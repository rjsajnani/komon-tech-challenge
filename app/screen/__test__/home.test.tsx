import React from 'react';

import Home from '../home';
import {
  renderWithProviders,
  RenderAPI,
  fireEvent,
} from '../../utils/testing-library';

describe('<Home />', () => {
  let tree: RenderAPI;

  beforeEach(function () {
    tree = renderWithProviders(<Home />);
  });

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  // this test was to check navigation (wasn't able to mock expo-router)

  // it('open the first link', async () => {
  //   fireEvent(await tree.findByText('Manage Social Media'), 'press');
  //   expect(navigation.screen).toBe([socialMedia])
  // });
});
