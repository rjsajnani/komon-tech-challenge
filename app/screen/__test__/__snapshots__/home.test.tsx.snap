// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    dataSet={{}}
    style={
      [
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 24,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 36,
          "textDecorationLine": undefined,
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    Komon Creator Dashboard
  </Text>
  <View
    dataSet={{}}
    style={
      [
        {},
        {
          "marginVertical": 16,
        },
      ]
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#262626",
          "marginBottom": 12,
          "marginTop": 12,
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          dataSet={{}}
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "black",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome5Free-Regular",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "400",
                },
              ]
            }
          >
            
          </Text>
          <View
            dataSet={{}}
            style={
              {
                "width": 12,
              }
            }
          />
          <View
            dataSet={{}}
            style={
              {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#1f2937",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Manage Social Media
              </Text>
            </View>
          </View>
          <View
            dataSet={{}}
            style={
              {
                "width": 12,
              }
            }
          />
          <View
            dataSet={{}}
            style={
              {
                "flexGrow": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#262626",
          "marginBottom": 12,
          "marginTop": 12,
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          dataSet={{}}
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "black",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome5Free-Brand",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "400",
                },
              ]
            }
          >
            
          </Text>
          <View
            dataSet={{}}
            style={
              {
                "width": 12,
              }
            }
          />
          <View
            dataSet={{}}
            style={
              {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#1f2937",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Community Messages
              </Text>
            </View>
          </View>
          <View
            dataSet={{}}
            style={
              {
                "width": 12,
              }
            }
          />
          <View
            dataSet={{}}
            style={
              {
                "flexGrow": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
